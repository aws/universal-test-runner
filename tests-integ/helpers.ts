// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/* eslint-disable no-console */

import { spawnSync } from 'child_process'
import path from 'path'
import fs from 'fs'

export function runSetupScript(adapter: string) {
  return runScript(adapter, 'setup.sh')
}

export function runCli(adapter: string, env: { [key: string]: string }) {
  // For some reason the executables don't get put into node_modules/.bin
  // when running in GitHub actions, so we us the JS file directly
  const EXECUTABLE = path.resolve(
    __dirname,
    '..',
    'packages',
    'universal-test-runner',
    'dist',
    'lib',
    'bin',
    'cli.js',
  )

  return runScript(adapter, 'run.sh', {
    ...env,
    RUN_TESTS: ['node', EXECUTABLE].join(' '),
  })
}

export function parseLogFile(adapter: string, logFileName: string) {
  const logs = JSON.parse(fs.readFileSync(path.resolve(getCwd(adapter), logFileName), 'utf-8'))
  logs.logs.forEach((log: any) => {
    log.timestamp = 'MOCK_VALUE'
  })
  return logs
}

export async function parseTestsToRun(adapter: string) {
  try {
    const config = await import(path.resolve(getCwd(adapter), 'config.json'))
    if (config && config.testsToRun) {
      return config.testsToRun
    }
  } catch (e: any) {
    // expect MODULE_NOT_FOUND if config file doesnt exist
    if (e.code !== 'MODULE_NOT_FOUND') {
      throw e
    }
  }
  return undefined
}

export async function parseTestsToRunWithFileAndSuite(adapter: string) {
  try {
    const config = await import(path.resolve(getCwd(adapter), 'config.json'))
    if (config && config.testsToRunWithFileAndSuite) {
      return config.testsToRunWithFileAndSuite
    }
  } catch (e: any) {
    // expect MODULE_NOT_FOUND if config file doesnt exist
    if (e.code !== 'MODULE_NOT_FOUND') {
      throw e
    }
  }
  return undefined
}

export function remove(adapter: string, filepath: string) {
  if (!filepath) {
    throw new Error('filepath must not be empty')
  }

  fs.rmSync(path.resolve(getCwd(adapter), filepath), {
    force: true,
    recursive: true,
  })
}

function getCwd(adapter: string) {
  return path.resolve(__dirname, 'test-projects', adapter)
}

function runScript(adapter: string, scriptName: string, env: { [key: string]: string } = {}) {
  const { status, stdout, stderr, error } = spawnSync('bash', [scriptName], {
    cwd: getCwd(adapter),
    stdio: process.env.DEBUG ? 'inherit' : 'pipe',
    env: {
      ...process.env,
      ...env,
    },
  })

  console.debug(
    `cwd = ${getCwd(adapter)}, 
    adapter = ${adapter}, 
    scriptName = ${scriptName}, 
    env = ${JSON.stringify(env)}`,
  )

  if (status !== 0) {
    console.log(stdout?.toString())
    console.error(stderr?.toString())
    console.error(error)
    throw new Error(
      `Script ${scriptName} failed to execute for adapter ${adapter}, exit code is ${status}`,
    )
  }

  return { status }
}
