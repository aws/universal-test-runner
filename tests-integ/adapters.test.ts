// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { runSetupScript, runCli, remove, parseLogFile, parseConfigFile } from './helpers'

const ADAPTERS = ['jest', 'pytest', 'maven', 'gradle', 'dotnet']

describe.each(ADAPTERS)('%s adapter', (adapter) => {
  beforeAll(() => {
    runSetupScript(adapter)
  })

  afterEach(() => {
    remove(adapter, 'logs')
  })

  it('runs all tests', () => {
    const { status } = runCli(adapter, {
      TEP_VERSION: '0.1.0',
      TEP_LOG_FILE_NAME: 'logs/logs.json',
    })
    expect(status).toBe(0)

    const logs = parseLogFile(adapter, 'logs/logs.json')
    expect(logs).toMatchSnapshot()
  })

  it('runs a subset of tests', () => {
    const { status } = runCli(adapter, {
      TEP_VERSION: '0.1.0',
      TEP_TESTS_TO_RUN: parseConfigFile(adapter) || 'test1|test2',
      TEP_LOG_FILE_NAME: 'logs/logs.json',
    })
    expect(status).toBe(0)

    const logs = parseLogFile(adapter, 'logs/logs.json')
    expect(logs).toMatchSnapshot()
  })
})
