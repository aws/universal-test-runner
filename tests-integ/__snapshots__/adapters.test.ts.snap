// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotnet adapter runs a subset of tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "Prime.UnitTests.Services.PrimeService_IsPrimeShould.IsPrime_InputIs0_ReturnFalse|Prime.UnitTests.Services.PrimeService_IsPrimeShould.IsPrime_InputIs1_ReturnFalse",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "dotnet",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`dotnet adapter runs all tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "dotnet",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`gradle adapter runs a subset of tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test1|test2",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "gradle",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`gradle adapter runs all tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "gradle",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`jest adapter runs a subset of tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test1|test2",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "jest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`jest adapter runs all tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "jest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`maven adapter runs a subset of tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test1|test2",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "maven",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`maven adapter runs all tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "maven",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`pytest adapter runs a subset of tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test1|test2",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "pytest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`pytest adapter runs a subset of tests 2`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test_example.py#TestSuite#test1",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "pytest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`pytest adapter runs a subset of tests 3`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test_example.py##test1",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "pytest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`pytest adapter runs a subset of tests 4`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_TESTS_TO_RUN": "test_example.py#TestSuite#test1",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "pytest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;

exports[`pytest adapter runs all tests 1`] = `
{
  "logs": [
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_READ_END",
    },
    {
      "data": {
        "TEP_LOG_FILE_NAME": "logs/logs.json",
        "TEP_VERSION": "0.1.0",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "DISCOVERED_PROTOCOL_ENV_VARS",
    },
    {
      "data": "0.1.0",
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "PROTOCOL_VERSION",
    },
    {
      "data": {
        "data": "pytest",
        "type": "ADAPTER_PATH",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_START",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "data": {
        "type": "ADAPTER_LOAD_END",
      },
      "level": "DEBUG",
      "timestamp": "MOCK_VALUE",
      "type": "CUSTOM",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_START",
    },
    {
      "level": "INFO",
      "timestamp": "MOCK_VALUE",
      "type": "TEST_RUN_END",
    },
  ],
}
`;
